### PR TITLE
Retire the C# 9 "What's new" tutorials and article.

### LINKED ISSUE #38033: Remove "What's new in C# 9"
After the C# 12 release, remove "What's new in C# 9". Move any appropriate tutorials from the "what's new" section to the regular tutorial section.

### PR DESCRIPTION
**Pull request by BillWagner**
Move all the C# 9 tutorials into the main "tutorials" folder, out from the "whats-new/tutorials" folder.

Retire the "What's new in C# 9" article, because it's not new. Move that content into the "C# Version history" article.

Fixes #38033 

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/architecture/microservices/microservice-ddd-cqrs-patterns/cqrs-microservice-reads.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/architecture/microservices/microservice-ddd-cqrs-patterns/cqrs-microservice-reads.md) | [Implement reads/queries in a CQRS microservice](https://review.learn.microsoft.com/en-us/dotnet/architecture/microservices/microservice-ddd-cqrs-patterns/cqrs-microservice-reads?branch=pr-en-us-38034) |
| [docs/core/deploying/trimming/incompatibilities.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/core/deploying/trimming/incompatibilities.md) | [Known trimming incompatibilities](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/trimming/incompatibilities?branch=pr-en-us-38034) |
| [docs/core/extensions/custom-configuration-provider.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/core/extensions/custom-configuration-provider.md) | [Implement a custom configuration provider in .NET](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/custom-configuration-provider?branch=pr-en-us-38034) |
| [docs/core/extensions/logging.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/core/extensions/logging.md) | [Assigns the env var to the value](https://review.learn.microsoft.com/en-us/dotnet/core/extensions/logging?branch=pr-en-us-38034) |
| [docs/core/tutorials/top-level-templates.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/core/tutorials/top-level-templates.md) | [docs/core/tutorials/top-level-templates](https://review.learn.microsoft.com/en-us/dotnet/core/tutorials/top-level-templates?branch=pr-en-us-38034) |
| [docs/core/whats-new/dotnet-5.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/core/whats-new/dotnet-5.md) | [What's new in .NET 5](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-5?branch=pr-en-us-38034) |
| [docs/csharp/language-reference/configure-language-version.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/csharp/language-reference/configure-language-version.md) | [C# language versioning - C# Guide](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version?branch=pr-en-us-38034) |
| [docs/csharp/tutorials/records.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/csharp/tutorials/records.md) | [Create record types](https://review.learn.microsoft.com/en-us/dotnet/csharp/tutorials/records?branch=pr-en-us-38034) |
| [docs/csharp/tutorials/top-level-statements.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/csharp/tutorials/top-level-statements.md) | [docs/csharp/tutorials/top-level-statements](https://review.learn.microsoft.com/en-us/dotnet/csharp/tutorials/top-level-statements?branch=pr-en-us-38034) |
| [docs/csharp/whats-new/csharp-version-history.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/csharp/whats-new/csharp-version-history.md) | [The history of C\#](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history?branch=pr-en-us-38034) |
| [docs/devops/create-dotnet-github-action.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/devops/create-dotnet-github-action.md) | [Rest of action.yml content removed for readability](https://review.learn.microsoft.com/en-us/dotnet/devops/create-dotnet-github-action?branch=pr-en-us-38034) |
| [docs/fundamentals/code-analysis/style-rules/ide0078-ide0260.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/fundamentals/code-analysis/style-rules/ide0078-ide0260.md) | ["IDE0078: Use pattern matching"](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0078-ide0260?branch=pr-en-us-38034) |
| [docs/fundamentals/code-analysis/style-rules/ide0083.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/fundamentals/code-analysis/style-rules/ide0083.md) | [Use pattern matching (`not` operator) (IDE0083)](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0083?branch=pr-en-us-38034) |
| [docs/fundamentals/code-analysis/style-rules/ide0084.md](https://github.com/dotnet/docs/blob/134fd88a4b17a2789a661b084d4976cbc53a3ebc/docs/fundamentals/code-analysis/style-rules/ide0084.md) | [docs/fundamentals/code-analysis/style-rules/ide0084](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0084?branch=pr-en-us-38034) |

</details>

### CODE REVIEW

**Comment by BillWagner** [comment]
Note for reviewers:

The only content edits were in the [history of C#](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history?branch=pr-en-us-38034). Other edits were moving and removing files, and updating files based on redirects.

File: docs/core/extensions/custom-configuration-provider.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Define a `Settings` record type entity for storing configuration values in the d
 
 :::code language="csharp" source="snippets/configuration/custom-provider/Models/Settings.cs":::
 
-For information on record types, see [Record types in C# 9](../../csharp/whats-new/csharp-9.md#record-types).
+For information on record types, see [Record types in C#](/dotnet/csharp/language-reference/builtin-types/records).
```
Comment (gewarren) [inline docs/core/extensions/custom-configuration-provider.md:29]:
```suggestion
For information on record types, see [Record types in C#](../../csharp/language-reference/builtin-types/records.md).
```

File: docs/fundamentals/code-analysis/style-rules/ide0078-ide0260.md [hunk 2]
```diff
@@ -126,6 +126,6 @@ For more information, see [How to suppress code analysis warnings](../suppress-w
 
 ## See also
 
-- [C# 9.0 pattern matching](../../../csharp/whats-new/csharp-9.md#pattern-matching-enhancements)
+- [C# 9.0 pattern matching](/dotnet/csharp/whats-new/csharp-version-history#c-version-9#pattern-matching-enhancements)
```
Comment (gewarren) [inline docs/fundamentals/code-analysis/style-rules/ide0078-ide0260.md:129]:
```suggestion
- [C# 9.0 pattern matching](../../../csharp/whats-new/csharp-version-history.md#c-version-9#pattern-matching-enhancements)
```

**Review by gewarren — APPROVED** [review]
LGTM

**Pull request merged**